src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ReadOnlySpanMarshaller.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Runtime.InteropServices.Marshalling
+{
+    [CLSCompliant(false)]
+    [CustomMarshaller(typeof(ReadOnlySpan<>), MarshalMode.ManagedToUnmanagedIn, typeof(ReadOnlySpanMarshaller<,>.ManagedToUnmanagedIn))]
+    [CustomMarshaller(typeof(ReadOnlySpan<>), MarshalMode.UnmanagedToManagedOut, typeof(ReadOnlySpanMarshaller<,>.UnmanagedToManagedOut))]
+    [ContiguousCollectionMarshaller]
+    public static unsafe class ReadOnlySpanMarshaller<T, TUnmanagedElement>
+        where TUnmanagedElement : unmanaged
+    {
+        public static unsafe class UnmanagedToManagedOut
+        {
+            public static TUnmanagedElement* AllocateContainerForUnmanagedElements(ReadOnlySpan<T> managed, out int numElements)
+            {
+                // Emulate the pinning behavior:
+                // If the span is over a null reference, then pass a null pointer.
+                if (Unsafe.IsNullRef(ref MemoryMarshal.GetReference(managed)))
```

Comment:
I assume we've documented this somewhere. Since this is a new marshaller it might be prudent to add this clearly to the design doc and once this is in, go back and add the `///` for the marshallers. Documentation for all these new types in source is a .NET 7 requirement for us.

Reply:
This was discussed in API review. I don't think it's in our design docs.

Reply:
I've added doc comments that reference this behavior